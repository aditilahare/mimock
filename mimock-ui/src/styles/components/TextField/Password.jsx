import React, { useState } from 'react';
import IconButton from '@mui/material/IconButton';
import OutlinedInput from '@mui/material/OutlinedInput';
import InputLabel from '@mui/material/InputLabel';
import InputAdornment from '@mui/material/InputAdornment';
import FormControl from '@mui/material/FormControl';
import Visibility from '@mui/icons-material/Visibility';
import VisibilityOff from '@mui/icons-material/VisibilityOff';

function PasswordField() {
	const [values, setValues] = useState({
		amount: '',
		password: '',
		weight: '',
		weightRange: '',
		showPassword: false,
	});

	const handleChange = (prop) => (event) => {
		setValues({ ...values, [prop]: event.target.value });
	};

	const handleClickShowPassword = () => {
		setValues({
			...values,
			showPassword: !values.showPassword,
		});
	};

	return (
		<FormControl
			data-testid='password-textfield'
			style={{ width: '80%' }}
			variant='outlined'
		>
			<InputLabel htmlFor='outlined-adornment-password'>Password</InputLabel>
			<OutlinedInput
				data-testid='outlined-password-input'
				id='outlined-adornment-password'
				type={values.showPassword ? 'text' : 'password'}
				value={values.password}
				onChange={handleChange('password')}
				endAdornment={
					<InputAdornment position='end'>
						<IconButton
							data-testid='show-password-icon'
							aria-label='toggle password visibility'
							onClick={handleClickShowPassword}
							edge='end'
						>
							<Choose>
								<When condition={values.showPassword}>
									<VisibilityOff />
								</When>
								<Otherwise>
									<Visibility />
								</Otherwise>
							</Choose>
						</IconButton>
					</InputAdornment>
				}
				label='Password'
			/>
		</FormControl>
	);
}

export default PasswordField;
