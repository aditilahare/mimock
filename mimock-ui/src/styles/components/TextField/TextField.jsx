import React from 'react';
import FormControl from '@mui/material/FormControl';
import InputLabel from '@mui/material/InputLabel';
import OutlinedInput from '@mui/material/OutlinedInput';
import FormHelperText from '@mui/material/FormHelperText';
import PropTypes from 'prop-types';

function BasicTextField({
	label,
	dataTestid,
	errorLabel,
	isError,
	inputValue,
	onChangeHandler,
	onBlurHandler,
	onFocusHandler,
}) {
	return (
		<FormControl data-testid={dataTestid} error={isError}>
			<InputLabel htmlFor='component-outlined'>{label}</InputLabel>
			<OutlinedInput
				data-testid='change-text-field'
				id='component-outlined'
				value={inputValue}
				onChange={onChangeHandler}
				onBlur={onBlurHandler}
				onFocus={onFocusHandler}
				label={label}
			/>

			<If condition={isError}>
				<FormHelperText id='component-error-text'>{errorLabel}</FormHelperText>
			</If>
		</FormControl>
	);
}

BasicTextField.propTypes = {
	label: PropTypes.string.isRequired,
	dataTestid: PropTypes.string.isRequired,
	inputValue: PropTypes.string.isRequired,
	onBlurHandler: PropTypes.func.isRequired,
	onChangeHandler: PropTypes.func.isRequired,
	onFocusHandler: PropTypes.func.isRequired,
	errorLabel: PropTypes.string,
	isError: PropTypes.bool,
};

export default BasicTextField;
