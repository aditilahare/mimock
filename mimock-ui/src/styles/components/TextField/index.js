import PasswordField from './PasswordField/PasswordField.jsx';
import UserNameField from './UserNameField/UserNameField.jsx';

export { UserNameField, PasswordField };
