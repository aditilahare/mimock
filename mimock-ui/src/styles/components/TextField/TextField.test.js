import React from 'react';
import { render } from '@testing-library/react';
import TextField from './TextField';

describe('Text Field', () => {
	let tree;

	it('should render basic text field', async () => {
		tree = await render(
			<TextField
				label='username'
				dataTestid='text-field'
				errorLabel='Username is required'
				isError={false}
				inputValue='Username'
				onChangeHandler={() => {
					console.log('Change handler');
				}}
				onBlurHandler={() => {
					console.log('Blur handler');
				}}
				onFocusHandler={() => {
					console.log('Focus handler');
				}}
			/>
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('text-field')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
