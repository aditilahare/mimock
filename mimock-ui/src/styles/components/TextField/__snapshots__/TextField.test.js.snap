// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Field should render basic text field 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    data-testid="text-field"
  >
    <label
      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="component-outlined"
    >
      username
    </label>
    <div
      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      data-testid="change-text-field"
    >
      <input
        aria-invalid="false"
        class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        id="component-outlined"
        type="text"
        value="Username"
      />
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-1z7n62"
        >
          <span>
            username
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
